### PR TITLE
[systemtest][upgrade] Attempt to deflake the upgrade tests and fix sending/receiving messages during upgrade

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix

### Description

In #3200 I totally forgot that the change of `--broker-list` to `--boostrap-server` can affect older versions of Kafka (during the upgrade) -> I tested it only with Kafka 2.5.0 and 2.6.0, problem is that `--boostrap-server` option was added to `2.5.0`. So I added ternary operator to determine which option use based on `Kafka` version.

Also this is an attempt to deflake the upgrade tests. I assume that main problem of the flakyness was this block of code:

```
if (KafkaTopicResource.kafkaTopicClient().inNamespace(NAMESPACE).withName(continuousTopicName).get() == null) {
                KafkaTopicResource.topic(CLUSTER_NAME, continuousTopicName, 3, 3, 2).done();
                // Add continuous topic to expectedTopicCount which will be check after upgrade procedures
                expectedTopicCount += 1;
            }
```
The problem is that if happen that topic is deleted and recreated, the `expectedTopicCount` will be increased and next check will not pass. This assumption is based on given result from PR run:

```
[WARNING] Flakes: 
[WARNING] io.strimzi.systemtest.upgrade.StrimziUpgradeST.testUpgradeStrimziVersion(JsonObject)[9]
[ERROR]   Run 1: StrimziUpgradeST.testUpgradeStrimziVersion:97->performUpgrade:351 KafkaTopic list doesn't have expected size
Expected: is <44>
     but: was <43>
[INFO]   Run 2: PASS
```

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=upgrade

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=upgrade

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java [hunk 1]
```diff
@@ -128,18 +128,19 @@ public VerifiableClient(VerifiableClientBuilder verifiableClientBuilder) {
 
         this.setAllowedArguments(this.clientType);
         this.clientArgumentMap = new ClientArgumentMap();
-        this.clientArgumentMap.put(ClientArgument.BOOTSTRAP_SERVER, bootstrapServer);
         this.clientArgumentMap.put(ClientArgument.TOPIC, topicName);
         this.clientArgumentMap.put(ClientArgument.MAX_MESSAGES, Integer.toString(maxMessages));
         if (kafkaUsername != null) this.clientArgumentMap.put(ClientArgument.USER,  kafkaUsername.replace("-", "_"));
 
+        String image = kubeClient().getPod(this.podName).getSpec().getContainers().get(0).getImage();
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java:135]: Yep, that's right, gonna take a look at it. Thanks

**Review by scholzj — APPROVED** [review]

**Comment by im-konge** [comment]
After discussion with @Frawless I'm gonna fix the azure pipelines for upgrade in next PR -> so this can be merge and all issues fixed.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/VerifiableClient.java:135]: Will be resolved in next PR.

**Review by Frawless — APPROVED** [review]

**Pull request merged**